readingDataPointsFilters:
  - RandomSamplingDataPointsFilter:
      prob: 0.75
  - BoundingBoxDataPointsFilter:
      xMin: -20.0
      xMax: 20.0
      yMin: -20.0
      yMax: 20.0
      
referenceDataPointsFilters:
  - SamplingSurfaceNormalDataPointsFilter:
      knn: 20
  - BoundingBoxDataPointsFilter:
      xMin: -20.0
      xMax: 20.0
      yMin: -20.0
      yMax: 20.0      

matcher:
  KDTreeMatcher:
    knn: 1

outlierFilters:
  - TrimmedDistOutlierFilter:
      ratio: 0.5

errorMinimizer:
  PointToPlaneErrorMinimizer

transformationCheckers:
  - CounterTransformationChecker:
      maxIterationCount: 40
  - DifferentialTransformationChecker:
      minDiffRotErr: 0.001
      minDiffTransErr: 0.01
      smoothLength: 4

inspector:
  VTKFileInspector:
     baseFileName : vissteps
     dumpDataLinks : 1
     dumpReading : 1
     dumpReference : 1

#logger:
  #FileLogger
